tests/integration/test_referral.cairo

```diff
@@ -0,0 +1,57 @@
+%lang starknet
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from src.interface.starknetid import StarknetId
+from src.interface.naming import Naming
+
+@external
+func __setup__() {
+    %{
+        from starkware.starknet.compiler.compile import get_selector_from_name
+        context.starknet_id_contract = deploy_contract("./lib/starknetid/src/StarknetId.cairo").contract_address
+        context.pricing_contract = deploy_contract("./src/pricing/main.cairo", [123]).contract_address
+        logic_contract_class_hash = declare("./src/naming/main.cairo").class_hash
+        context.naming_contract = deploy_contract("./lib/cairo_contracts/src/openzeppelin/upgrades/presets/Proxy.cairo", [logic_contract_class_hash,
+            get_selector_from_name("initializer"), 4, 
+            context.starknet_id_contract, context.pricing_contract, 456, 0]).contract_address
+    %}
+    return ();
+}
+
+@external
+func test_simple_buy_without_sponsor{syscall_ptr: felt*, range_check_ptr, pedersen_ptr: HashBuiltin*}() {
```

Comment:
Why do we need to test it considering that we tested it before ?

Reply:
I added a test to ensure that what I added in the code worked but I can remove it